### PR TITLE
no-issue: Split DataBroker pullMetadata API per type

### PR DESCRIPTION
**Pull request by kael89**
Splitting `DataBroker.pullMetadata` into two APIs, one for data elements and one for data group.

Splitting the Service APIs will be done in another PR

### CODE REVIEW

File: packages/data-broker/src/__tests__/DataBroker/DataBroker.test.ts [hunk 1]
```diff
@@ -166,35 +172,71 @@ describe('DataBroker', () => {
...
-      const mapping = new DataServiceMapping([
-        { dataSource: DATA_ELEMENTS.DHIS_01, service_type: 'dhis', config: {} },
-      ]);
 
-      const testData: [MethodUnderTest, any[]][] = [
-        ['push', [{ code: 'DHIS_01', type: 'dataElement' }, [{ value: 2 }], TO_OPTIONS]],
-        ['delete', [{ code: 'DHIS_01', type: 'dataElement' }, { value: 2 }, TO_OPTIONS]],
-        ['pullMetadata', [{ code: 'DHIS_01', type: 'dataElement' }, TO_OPTIONS]],
-      ];
+      it('passes mapping to service: push', async () => {
```
Comment (kael89) [inline packages/data-broker/src/__tests__/DataBroker/DataBroker.test.ts:176]: This section is just breaking apart the parameterised tests into separate test blocks per API

I think parameterised tests should only be used for simple, pure functions where we want to test multiple variations for the same API, e.g. maths, date handling etc. They are probably overused in `tupaia` (guilty as charged!)

File: packages/aggregator/global.d.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
```
Comment (edmofro) [inline packages/aggregator/global.d.ts:3]:
```suggestion
 * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
```

File: packages/data-broker/.env.example [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Used in tests
+DB_NAME=
+DB_PASSWORD=
+DB_PORT=
+DB_URL=
+DB_USER=
```
Comment (edmofro) [inline packages/data-broker/.env.example:6]: Thanks for adding the example

**Review by edmofro — APPROVED** [review]
All lgtm!! Thank you as always @kael89

**Comment by kael89** [comment]
Can I merge this @edmofro?

**Comment by edmofro** [comment]
Yes go for it!

**Pull request merged**